pkgs/build-support/docker/default.nix

```diff
@@ -835,179 +835,161 @@ rec {
-        baseJson = writeText "${baseName}-base.json" (builtins.toJSON {
-          inherit config;
-          architecture = defaultArch;
-          os = "linux";
-        });
-
-        contentsList = if builtins.isList contents then contents else [ contents ];
-
-        # We store the customisation layer as a tarball, to make sure that
-        # things like permissions set on 'extraCommands' are not overriden
-        # by Nix. Then we precompute the sha256 for performance.
-        customisationLayer = symlinkJoin {
-          name = "${baseName}-customisation-layer";
-          paths = contentsList;
-          inherit extraCommands fakeRootCommands;
-          nativeBuildInputs = [
-            fakeroot
-          ] ++ optionals enableFakechroot [
-            fakechroot
-            # for chroot
-            coreutils
-            # fakechroot needs getopt, which is provided by util-linux
-            util-linux
-          ];
-          postBuild = ''
-            mv $out old_out
-            (cd old_out; eval "$extraCommands" )
-
-            mkdir $out
-            ${optionalString enableFakechroot ''
-              export FAKECHROOT_EXCLUDE_PATH=/dev:/proc:/sys:${builtins.storeDir}:$out/layer.tar
-            ''}
-            ${optionalString enableFakechroot ''fakechroot chroot $PWD/old_out ''}fakeroot bash -c '
-              source $stdenv/setup
-              ${optionalString (!enableFakechroot) ''cd old_out''}
-              eval "$fakeRootCommands"
-              tar \
-                --sort name \
-                --numeric-owner --mtime "@$SOURCE_DATE_EPOCH" \
-                --hard-dereference \
-                -cf $out/layer.tar .
-            '
-
-            sha256sum $out/layer.tar \
-              | cut -f 1 -d ' ' \
-              > $out/checksum
-          '';
-        };
-
-        closureRoots = lib.optionals includeStorePaths /* normally true */ (
-          [ baseJson customisationLayer ]
-        );
-        overallClosure = writeText "closure" (lib.concatStringsSep " " closureRoots);
-
-        # These derivations are only created as implementation details of docker-tools,
-        # so they'll be excluded from the created images.
-        unnecessaryDrvs = [ baseJson overallClosure customisationLayer ];
-
-        conf = runCommand "${baseName}-conf.json"
-          {
-            inherit fromImage maxLayers created;
-            imageName = lib.toLower name;
-            passthru.imageTag =
-              if tag != null
-              then tag
-              else
-                lib.head (lib.strings.splitString "-" (baseNameOf conf.outPath));
-            paths = buildPackages.referencesByPopularity overallClosure;
-            nativeBuildInputs = [ jq ];
-          } ''
-          ${if (tag == null) then ''
-            outName="$(basename "$out")"
-            outHash=$(echo "$outName" | cut -d - -f 1)
+    let
+      baseImageLayersCount = if fromImage == null then 0 else
+      import (
```

Comment:
This is an import from derivation, which isn't necessary. Could you move the layer count related logic back into the builder(s)?

Reply:
will do

Reply:
fixed in 0c8d1530a545c6ff8ebfe740f6cf83bd710b7138